---
title: "Logbook Jasper Jonker"
author: "Jasper Jonker"
date: "2024-16-10"
output: html_document
editor_options: 
  markdown: 
    wrap: sentence
---
## deseq 
Hieronder staat code voor de verwerking van deseq:

Dit is van teams door ramon.

```{r}
file.names <- list.files('/students/2024-2025/Thema05/BlaasKanker/Transcriptomics/output/mapping/',
                         pattern = '*_star_ReadsPerGene.out.tab')

 

## Function for reading in files
read_sample <- function(file.name) {
  ## Extract the sample name for naming the column (retaining the 'SRR....' part)
  sample.name <- strsplit(file.name, ".", fixed = TRUE)[[1]][1]
  sample <- read.table(file.name, header = FALSE, sep="\t", 
                       row.names = NULL, skip = 4)
  ## Rename the count column
  names(sample)[2] <- sample.name
  ## Return a subset containing the transcript ID and sample name columns
  return(sample[c(1, 2)])
}

 

```

```{r}

 

setwd('/students/2024-2025/Thema05/BlaasKanker/Transcriptomics/output/mapping/')

 

## Read the FIRST sample
counts <- read_sample(file.names[1])

 

## Read the remaining files and merge the contents
for (file.name in file.names[2:length(file.names)]) {
  sample <- read_sample(file.name)
  counts <- merge(counts, sample, by = 1)
}

 

# Set the row names to the transcript IDs
rownames(counts) <- counts$V1
counts <- counts[-1]

```

```{r}

 

col_data <- read.csv("/students/2024-2025/Thema05/BlaasKanker/etc/DSEQ_verwerking(Sheet1).csv", sep = ";")
rownames(col_data) <- col_data[,1]
col_data
```

## Annot count_data
Het annoteren van onze data zorgt ervoor dat de genen de NCBI naamgeving krijgen. Zo is het straks overzichtelijker om informatie te vinden over de genen.


```{r}
library(dplyr)
library(tidyr)

 


col_data <- col_data %>%
    dplyr::group_by(strain, treatment) %>%
    dplyr::mutate(r_num = row_number()) %>%
    dplyr::ungroup() %>%
    mutate(condition = paste0(strain, "_", treatment, "_r", r_num))
head(col_data)

 

```

In de bestanden kon niks gevonden worden over de genen, maar zo is het wel duidelijker waar elke kolom voor staat.
Dat moet nu nog toegepast worden in de counts df
```{r}

 

for (i in 1:nrow(col_data)) {
  idx <- grep(col_data$Run[i], names(counts))
  names(counts)[idx] <- col_data$condition[i]
}
head(counts)
```
Nu zijn de SRR* namen vervangen met waar ze voor staan en is de df duidelijker te lezen, om straks makkelijker de kolommen op te halen die horen bij elk mogelijke variant groepeer ik deze.

 

```{r}
print(colnames(counts))
C57BL_Vehicle <- grep("C57BL/6_Vehicle", names(counts))
C57BL_Entinostat <- grep("C57BL/6_Entinostat", names(counts))
C57BL_Baseline <- grep("C57BL/6_Baseline", names(counts))

 

nsg_Vehicle <- grep("NSG_Vehicle", names(counts))
nsg_Entinostat <- grep("NSG_Entinostat", names(counts))
nsg_Baseline <- grep("NSG_Baseline", names(counts))
```
Nu kunnen deze variabelen gebruikt worden om de juiste kolommen te selecteren.

 

Dan kan dit nu samengevoegd worden met DESEQ, wat Janine heeft uitgezocht. Bekijk haar logboek voor meer informatie over de code.

```{r}
# Making deseqdataset with counts en coldata
dds <- DESeqDataSetFromMatrix(countData = counts,
                              colData = col_data,
                              design = ~ 0 + source_name)
head(dds$source_name)
```

 

```{r}
# Prefiltering on low gene counts
keep <- rowSums(counts(dds)) >= 10
ddst <- dds[keep,]
 
```

 

```{r}
# Setting factor level
dds$treatment <- relevel(dds$treatment, ref = "Baseline")
dds$treatment <- factor(dds$treatment, levels = c("Baseline","Entinostat", "Vehicle"), )
dds$treatment
```

 

```{r}
# Running deseq
dds <- DESeq(dds)
res <- results(dds)

```

```{r}
head(res)
```

Dit is goed maar niet wat ik wil gebruiken. Mu word er NSG_bladder tumor_Vehicle vs C57BL/6_bladder tumor_Baseline  tegenover elkaar gezet en moet ik noet doen. Ik moet de C57BL/6 met elkaar vergelijken. 
In de quickstrart staat nog lfcShrink doen.



```{r}
lfc_entinostat <- lfcShrink(dds, coef="source_nameC57BL/6_bladder.tumor_Entinostat", type="apeglm")

lfc_basline <- lfcShrink(dds, coef="source_nameC57BL/6_bladder tumor_Baseline", type="apeglm")

```



## Visualization oefen

Ik ga alvast bezig om te kijken hoe we onze data van trimmotic zo dat we in 1 keer door kunnen gaan met alles want we hebben maar een korte tijd om het allemaal te laten zien 
### Pathway Analysis oefen
We gaan zo ie zo wel Pathway analysis doen.
In deze grafiek kunnen we gaan kijken welke gevolgen de varianten hebben later op in het gebied van de biologische processen.

16-10 <https://pathview.r-forge.r-project.org/pathview.pdf> is een een pdf met alle informatie

```{r, echo=FALSE}
library(pathview)
data(gse16873.d)
pv.out <- pathview(gene.data = gse16873.d[, 1], pathway.id = "04110",
 species = "hsa", out.suffix = "gse16873")
```

![](/homes/jjonker2/Documents/GenomicsTranscriptomics/Transcriptomics/logbooks/hsa04110.gse16873.png)
```{r}
library(pathview)
data(gse16873.d)

```

### Volcano plot oefen 

17-10 
#### manier 1 
Dit is 1 manier om het te doen.

```{r}
tmp <- readRDS("/students/2024-2025/Thema05/BlaasKanker/Transcriptomics/testData/de_df_for_volcano.rds")
de <- tmp[complete.cases(tmp), ]
de$delabel <- NA
de$diffexpressed <- "NO"
de$delabel[de$diffexpressed != "NO"] <- de$gene_symbol[de$diffexpressed != "NO"]

de$diffexpressed[de$log2FoldChange > 0.6 & de$pvalue < 0.05] <- "UP"
# if log2Foldchange < -0.6 and pvalue < 0.05, set as "DOWN"
de$diffexpressed[de$log2FoldChange < -0.6 & de$pvalue < 0.05] <- "DOWN"

ggplot(data=de, aes(x=log2FoldChange, y=-log10(pvalue), col=diffexpressed, label=delabel)) + 
    geom_point() + 
    theme_minimal() +
    geom_text() +
    geom_vline(xintercept=c(-0.6, 0.6), col="red") +
    geom_hline(yintercept=-log10(0.05), col="red") +
    scale_color_manual(values=c("blue", "black", "purple"))
```

#### manier2

```{r}
library(airway)
library(magrittr)
data('airway')
airway$dex %<>% relevel('untrt')

ens <- rownames(airway)
```

```{r}
library(org.Hs.eg.db)
symbols <- mapIds(org.Hs.eg.db, keys = ens,
    column = c('SYMBOL'), keytype = 'ENSEMBL')
symbols <- symbols[!is.na(symbols)]
symbols <- symbols[match(rownames(airway), names(symbols))]
rownames(airway) <- symbols
keep <- !is.na(rownames(airway))
airway <- airway[keep,]
```

```{r}
library('DESeq2')

dds <- DESeqDataSet(airway, design = ~ cell + dex)

dds <- DESeq(dds, betaPrior=FALSE)

res <- results(dds,
contrast = c('dex','trt','untrt'))

res_1 <- lfcShrink(dds,
contrast = c('dex','trt','untrt'), res=res, type = 'normal')

```

```{r}
  EnhancedVolcano(res_1,
    lab = rownames(res_1),
    x = 'log2FoldChange',
    y = 'pvalue')
```
### Heatmap oefen




###Clusterprofiling gsa oefen 


http://yulab-smu.top/biomedical-knowledge-mining-book/enrichment-overview.html#gsea-algorithm 
```{r}
library("clusterProfiler")

data(geneList, package="DOSE")
head(geneList)
```

```{r}
gene <- names(geneList)[abs(geneList) > 2]
head(gene)
```
GO classification
```{r}
ggo <- groupGO(gene     = gene,
               OrgDb    = org.Hs.eg.db,
               ont      = "CC",
               level    = 3,
               readable = TRUE)

head(ggo)
```

GO over-representation analysis

Hier kan je zien dat bepaalde genen een over presentaie hebben omdat ze een p.dajust hebben kleiner dan 0.05 en qvaule kleiner dan 0,05 dat betekent dat de deze genen meer tot uiting komen dan verwacht wordt.
https://www.youtube.com/watch?v=egO7Lt92gDY 
```{r}
ego <- enrichGO(gene          = gene,
                universe      = names(geneList),
                OrgDb         = org.Hs.eg.db,
                ont           = "CC",
                pAdjustMethod = "BH",
                pvalueCutoff  = 0.01,
                qvalueCutoff  = 0.05,
        readable      = TRUE)
head(ego)
```

GO Gene Set Enrichment Analysis


https://www.youtube.com/watch?v=egO7Lt92gDY 
https://www.youtube.com/watch?v=Yi4d7JIlAsM

Dus je ziet hier GO:0000775 het meest interessant is door nes met een score van 0.6230073
	

```{r}
ego3 <- gseGO(geneList     = geneList,
              OrgDb        = org.Hs.eg.db,
              ont          = "CC",
              minGSSize    = 100,
              maxGSSize    = 500,
              pvalueCutoff = 0.05,
              verbose      = FALSE)
head(ego3)
```

```{r}
p1 <- gseaplot(ego3, geneSetID = 1, by = "runningScore", title = ego3$Description[1])
p2 <- gseaplot(ego3, geneSetID = 1, by = "preranked", title = ego3$Description[1])
cowplot::plot_grid(p1, p2, ncol=1, labels=LETTERS[1:3])
```
https://yulab-smu.top/biomedical-knowledge-mining-book/enrichplot.html




## Visualization
###Clusterprofiling gsa

```{r}
head(res)
```

### Volcano 
```{r}

 EnhancedVolcano(res,
    lab = rownames(res),
    x = 'log2FoldChange',
    y = 'pvalue',
    title = "behandeld tegen onbehandeld",
     pCutoff = 10e-16,
    FCcutoff = 1.5,
    pointSize = 3.0,
    labSize = 6.0,)

  
```
Hoogste 20 
```{r}
resultaat_oder <- res[order(res$padj), ]
top20genen <- rownames(head(resultaat_oder, 20))
EnhancedVolcano(res,
                lab = rownames(res),
                x = 'log2FoldChange',
                y = 'padj',
                selectLab = top20genen,
                title = "behandeld tegen onbehandeld",
                pCutoff = 1,
                pointSize = 2.5,
                labSize = 4
                )
```

### MA 
```{r}
plotMA(res)
```
### pheatmap

```{r}
library(pheatmap)
topgene <- head(order(res$padj), 20 )
mat <- assay(dds)[topgene, ]

pheatmap(mat, cluster_rows=TRUE, cluster_cols = TRUE)
```

### PCA

```{r}
rld <- rlog(dds)

plotPCA(rld)
```

